File: 0903/903.py
class Solution:
    def valid_permutations_for_di_sequence(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.valid_permutations_for_di_sequence(root.left), self.valid_permutations_for_di_sequence(root.right))

File: 0903/903.go
func validPermutationsForDiSequence(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := validPermutationsForDiSequence(root.Left)
	r := validPermutationsForDiSequence(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}